xds/internal/client/helper_test.go

```diff
@@ -193,183 +198,104 @@ func TestNewConfig(t *testing.T) {
 		}
 		return nil, os.ErrNotExist
 	}
-	oldOnceDoerFunc := onceDoerFunc
-	onceDoerFunc = func(f func()) {
-		// Disable the synce.Once functionality to read the bootstrap file.
-		// Instead, read it everytime NewConfig() is called so that we can test
-		// with different file contents.
-		f()
-	}
 	defer func() {
 		fileReadFunc = oldFileReadFunc
-		onceDoerFunc = oldOnceDoerFunc
 		os.Unsetenv(bootstrapFileEnv)
 	}()
 
 	tests := []struct {
-		name             string
-		fName            string
-		wantErr          bool
-		wantBalancerName string
-		wantNodeProto    *basepb.Node
-		// TODO: It doesn't look like there is an easy way to compare the value
-		// stored in Creds with an expected value. Figure out a way to make it
-		// testable.
+		name       string
+		fName      string
+		wantConfig *Config
 	}{
 		{
-			name:             "non-existent-bootstrap-file",
-			fName:            "dummy",
-			wantErr:          true,
-			wantBalancerName: "",
-			wantNodeProto:    nil,
+			name:       "non-existent-bootstrap-file",
+			fName:      "dummy",
```

Comment:
Use `name` as the file name, instead of a different `fName`?

Reply:
Done.